qiskit/visualization/latex.py

```diff
@@ -345,690 +321,303 @@ def _get_beamer_page(self):
 
         return (height, width, self.scale)
 
-    def _get_mask(self, creg_name):
-        mask = 0
-        for index, cbit in enumerate(self.clbit_list):
-            if creg_name == cbit.register:
-                mask |= (1 << index)
-        return mask
+    def _get_gate_ctrl_text(self, op):
+        """Load the gate_text and ctrl_text strings based on names and labels"""
+        op_label = getattr(op.op, 'label', None)
+        op_type = type(op.op)
+        if hasattr(op.op, 'base_gate'):
+            base_name = op.op.base_gate.name
+            base_label = op.op.base_gate.label
+            base_type = type(op.op.base_gate)
+        else:
+            base_name = base_label = base_type = None
+        ctrl_text = None
+
+        if base_label:
+            gate_text = base_label
+            ctrl_text = op_label
+        elif op_label and isinstance(op.op, ControlledGate):
+            gate_text = base_name
+            ctrl_text = op_label
+        elif op_label:
+            gate_text = op_label
+        elif base_name:
+            gate_text = base_name
+        else:
+            gate_text = op.name
+
+        if gate_text in self._style['disptex']:
+            gate_text = self._style['disptex'][gate_text]
+            # Only add mathmode formatting if not already mathmode in disptex
+            if gate_text[0] != '$' and gate_text[-1] != '$':
```

Comment:
We should standarize `self._style` to avoid "guessing" if math mode should be used or not.

Reply:
Also agreed for new issue.